router/usecase/pools/routable_cw_alloy_transmuter_pool.go

```diff
@@ -0,0 +1,169 @@
+// - the token in amount is greater than the balance of the token out
+//
+// Note that balance validation does not apply to alloyed asset since it can be minted or burned by the pool.
+func (r *routableAlloyTransmuterPoolImpl) CalculateTokenOutByTokenIn(ctx context.Context, tokenIn sdk.Coin) (sdk.Coin, error) {
+	tokenOutAmt, err := r.CalcTokenOutAmt(tokenIn, r.TokenOutDenom)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	tokenOutAmtInt := osmomath.NewIntFromBigInt(tokenOutAmt.TruncateInt().BigInt())
+
+	// Validate token out balance if not alloyed
+	if r.TokenOutDenom != r.AlloyTransmuterData.AlloyedDenom {
+		if err := validateTransmuterBalance(tokenOutAmtInt, r.Balances, r.TokenOutDenom); err != nil {
+			return sdk.Coin{}, err
+		}
+	}
+
+	return sdk.Coin{Denom: r.TokenOutDenom, Amount: tokenOutAmtInt}, nil
+}
+
+// GetTokenOutDenom implements RoutablePool.
+func (r *routableAlloyTransmuterPoolImpl) GetTokenOutDenom() string {
+	return r.TokenOutDenom
+}
+
+// String implements sqsdomain.RoutablePool.
+func (r *routableAlloyTransmuterPoolImpl) String() string {
+	return fmt.Sprintf("pool (%d), pool type (%d) Transmuter with alloyed denom, pool denoms (%v), token out (%s)", r.ChainPool.PoolId, poolmanagertypes.CosmWasm, r.GetPoolDenoms(), r.TokenOutDenom)
+}
+
+// ChargeTakerFeeExactIn implements sqsdomain.RoutablePool.
+// Returns tokenInAmount and does not charge any fee for transmuter pools.
+func (r *routableAlloyTransmuterPoolImpl) ChargeTakerFeeExactIn(tokenIn sdk.Coin) (inAmountAfterFee sdk.Coin) {
+	return tokenIn
+}
+
+// GetTakerFee implements sqsdomain.RoutablePool.
+func (r *routableAlloyTransmuterPoolImpl) GetTakerFee() math.LegacyDec {
+	return r.TakerFee
+}
+
+// SetTokenOutDenom implements sqsdomain.RoutablePool.
+func (r *routableAlloyTransmuterPoolImpl) SetTokenOutDenom(tokenOutDenom string) {
+	r.TokenOutDenom = tokenOutDenom
+}
+
+// CalcSpotPrice implements sqsdomain.RoutablePool.
+func (r *routableAlloyTransmuterPoolImpl) CalcSpotPrice(ctx context.Context, baseDenom string, quoteDenom string) (osmomath.BigDec, error) {
+	return r.CalcTokenOutAmt(sdk.Coin{Denom: baseDenom, Amount: osmomath.OneInt()}, quoteDenom)
+}
+
+// IsGeneralizedCosmWasmPool implements sqsdomain.RoutablePool.
+func (*routableAlloyTransmuterPoolImpl) IsGeneralizedCosmWasmPool() bool {
+	return false
+}
+
+// GetCodeID implements sqsdomain.RoutablePool.
+func (r *routableAlloyTransmuterPoolImpl) GetCodeID() uint64 {
+	return r.ChainPool.CodeId
+}
+
+// FindNormalizationFactors finds the normalization factors for the given token in and token out denoms.
+// It is required for calculating token out & spot price.
+// For more information about normalization factor, please refer to [transmuter documentation](https://github.com/osmosis-labs/transmuter/tree/v3.0.0?tab=readme-ov-file#normalization-factors).
+func (r *routableAlloyTransmuterPoolImpl) FindNormalizationFactors(tokenInDenom, tokenOutDenom string) (osmomath.Int, osmomath.Int, error) {
```

Comment:
I like how you added a separate test for a lower-level calc method but could we also unit-test this one?

Seems like a few things are happening here, and we would definitely benefit from strict coverage